inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/project/EvaluationSimulationPanel.java

```diff
@@ -0,0 +1,218 @@
+import de.tudarmstadt.ukp.inception.recommendation.api.evaluation.EvaluationResult;
+import de.tudarmstadt.ukp.inception.recommendation.api.evaluation.IncrementalSplitter;
+import de.tudarmstadt.ukp.inception.recommendation.api.model.Recommender;
+import de.tudarmstadt.ukp.inception.recommendation.api.recommender.RecommendationEngine;
+import de.tudarmstadt.ukp.inception.recommendation.api.recommender.RecommendationEngineFactory;
+import de.tudarmstadt.ukp.inception.recommendation.api.recommender.RecommendationException;
+import de.tudarmstadt.ukp.inception.recommendation.chart.ChartPanel;
+import de.tudarmstadt.ukp.inception.recommendation.model.LearningCurve;
+
+public class EvaluationSimulationPanel
+    extends Panel
+{
+    private static final long serialVersionUID = 4306746527837380863L;
+
+    private static final String MID_CHART_CONTAINER = "chart-container";
+    private static final String MID_SIMULATION_START_BUTTON = "simulation-start-button";
+    private static final String MID_FORM = "form";
+    
+    private static final double TRAIN_PERCENTAGE = 0.8;
+    private static final int INCREMENT = 250;
+    private static final int LOW_SAMPLE_THRESHOLD = 10;
+
+    private static final Logger LOG = LoggerFactory.getLogger(EvaluationSimulationPanel.class);
+    
+    private @SpringBean DocumentService documentService;
+    private @SpringBean UserDao userDao;
+    private @SpringBean RecommenderFactoryRegistry recommenderRegistry;
+
+    private final Panel emptyPanel;
+    private final Project project;
+    private final IModel<Recommender> selectedRecommenderPanel;
+
+    public EvaluationSimulationPanel(String aId, Project aProject,
+            IModel<Recommender> aSelectedRecommenderPanel)
+    {
+        super(aId);
+        project = aProject;
+        selectedRecommenderPanel = aSelectedRecommenderPanel;
+                
+        Form<Recommender> form = new Form<>(MID_FORM);
+        add(form);
+        
+        emptyPanel  = new EmptyPanel(MID_CHART_CONTAINER);
+        emptyPanel.setOutputMarkupPlaceholderTag(true);
+        emptyPanel.setOutputMarkupId(true);
+        form.add(emptyPanel);
+        
+        // clicking the start button the annotated documents are evaluated and the learning curve
+        // for the selected recommender is plotted in the hCart Panel
+        @SuppressWarnings({ "unchecked", "rawtypes" })
+        LambdaAjaxButton startButton = new LambdaAjaxButton(MID_SIMULATION_START_BUTTON,
+            (_target, _form) -> {
+                 
+                    // replace the empty panel with chart panel on click event so the chard renders
+                    // with the loadable detachable model.
+                ChartPanel chartPanel = new ChartPanel(MID_CHART_CONTAINER, 
+                        LoadableDetachableModel.of(this::renderChart));
+                chartPanel.setOutputMarkupPlaceholderTag(true);
+                chartPanel.setOutputMarkupId(true);
+                
```

Comment:
@Sibgha360 I found out that you can trigger the data loading by adding `chartPanel.getDefaultModel().getObject();` here. It still lags a bit, but shows the spinner for some time.